src/history/hs.c

```diff
@@ -575,6 +585,8 @@ __wt_hs_insert_updates(WT_CURSOR *cursor, WT_BTREE *btree, WT_PAGE *page, WT_MUL
         /* Skip TOMBSTONE at the end of the update chain. */
         if (upd->type == WT_UPDATE_TOMBSTONE) {
             if (modifies.size > 0) {
+                if (upd->start_ts == WT_TS_NONE)
```

Comment:
This isn't great since we're going to `search_near` on every non-timestamped tombstone that we see. We can probably do something more clever like wait until the next insert (that may be timestamped) and then delete backwards then forwards? I think this is functionally correct though so we should address it if/when it causes perf issues.